modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/security/IgniteSessionRepository.java

```diff
@@ -117,12 +157,32 @@ public IgniteSessionRepository setDefaultMaxInactiveInterval(int dfltMaxInactive
     }
 
     /** {@inheritDoc} */
-    @Override public void delete(String id) {
+    @Override public void delete(String sesId) {
         try {
-            cache().remove(id);
+            txMgr.doInTransaction(() -> {
+                Set<String> accEmails = sesToAccIdx.get(sesId);
+
+                sesToAccIdx.delete(sesId);
+
+                accEmails.forEach(sesToAccIdx::delete);
+
+                sessionsCache.cache().remove(sesId);
+            });
         }
         catch (RuntimeException e) {
             throw convertToDatabaseNotAvailableException(e, messages.getMessage("err.db-not-available"));
         }
     }
+
+    /** {@inheritDoc} */
+    @Override public Map<String, ExpiringSession> findByIndexNameAndIndexValue(String idxName, String idxVal) {
+        if (!PRINCIPAL_NAME_INDEX_NAME.equals(idxName))
+            return Collections.emptyMap();
+
+        return accToSesIdx
+            .get(idxVal)
+            .stream()
+            .map(this::getSession)
+            .collect(toMap(Session::getId, ses -> ses));
```

Comment:
Use Function#identity instead of ses -> ses

Reply:
Fixed.